test/ERC6538Registry.t.sol

```diff
@@ -79,260 +132,184 @@ contract RegisterKeys is ERC6538RegistryTest {
-      );
-      assertEq(
-        registry.stealthMetaAddressOf(address(erc1271MockContract), schemeId), stealthMetaAddress
-      );
+      registry.registerKeysOnBehalf(registrant, schemeId, signature, stealthMetaAddress);
+      assertEq(registry.stealthMetaAddressOf(registrant, schemeId), stealthMetaAddress);
     }
   }
 
-  function testFuzz_EmitsStealthMetaAddressSetEventsForTheSameRegistrantAtLeastTwiceWhenProvidedAValidErc1271Signature(
-    string memory name,
+  function testFuzz_UpdatesTheStealthMetaAddressForAnExistingRegistrantWhenProvidedAValidErc1271Signature(
     uint256 schemeId,
     bytes memory stealthMetaAddress,
+    bytes memory signature,
     uint256 numOfUpdates
   ) external {
-    numOfUpdates = bound(numOfUpdates, 1, 50);
+    numOfUpdates = bound(numOfUpdates, 2, 50);
 
-    (address alice, uint256 alicePk) = makeAddrAndKey(name);
-
-    vm.prank(alice);
-    ERC1271MockContract erc1271MockContract = new ERC1271MockContract();
+    MockERC1271Signer mockRegistrant = new MockERC1271Signer();
+    mockRegistrant.setResponse__isValidSignature(true);
 
     for (uint256 nonce = 0; nonce < numOfUpdates; nonce++) {
-      SigUtils.RegistrantInfo memory registrantInfo =
-        SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, nonce);
-      (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, sigUtils.getTypedDataHash(registrantInfo));
-      bytes memory signature = abi.encodePacked(r, s, v);
-
-      vm.expectEmit(true, true, true, true);
-      emit StealthMetaAddressSet(address(erc1271MockContract), schemeId, stealthMetaAddress);
       registry.registerKeysOnBehalf(
-        address(erc1271MockContract), schemeId, signature, stealthMetaAddress
+        address(mockRegistrant), schemeId, signature, stealthMetaAddress
       );
+      assertEq(registry.stealthMetaAddressOf(address(mockRegistrant), schemeId), stealthMetaAddress);
     }
   }
 
-  function testFuzz_RevertIf_AnErc712SignatureFromAnAddressIsDifferentFromTheOneBeingRegistered(
-    string memory name,
-    address bob,
+  function testFuzz_RevertIf_TheDataIsErc712SignedByAnAddressOtherThanTheRegistrant(
+    string memory seed,
+    address registrant,
     uint256 schemeId,
     bytes memory stealthMetaAddress
   ) external {
-    ( /*address alice */ , uint256 alicePk) = makeAddrAndKey(name);
-    SigUtils.RegistrantInfo memory registrantInfo =
-      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
-    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
-    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
-    bytes memory signature = abi.encodePacked(r, s, v);
+    (address notRegistrant, uint256 notRegistrantPrivateKey) = makeAddrAndKey(seed);
+    vm.assume(notRegistrant != registrant);
+    bytes memory signature =
+      _generateRegistrationSignature(notRegistrantPrivateKey, schemeId, stealthMetaAddress, 0);
 
     vm.expectRevert();
```

Comment:
Just got a comment from matt we might want to implement wherever we have `vm.expectRevert()`. Comment:https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5#discussion_r1476213212

Reply:
Updated here [fc2a267](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/7/commits/fc2a267daa35427360d802230678342f94ba3bbc)